src/tracking/lib/amrit-tracking.service.ts

```diff
@@ -0,0 +1,63 @@
+import { Injectable, Inject } from '@angular/core';
+import { TrackingProvider } from './tracking-provider';
+import { SessionStorageService } from '../../registrar/services/session-storage.service';
+import { Router, NavigationEnd } from '@angular/router';
+import { filter } from 'rxjs/operators';
+import { TRACKING_PROVIDER } from './tracking.tokens';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AmritTrackingService {
+  constructor(
+    @Inject(TRACKING_PROVIDER) private trackingProvider: TrackingProvider,
+    private sessionStorage: SessionStorageService,
+    private router: Router
+  ) {
+    // Initialise underlying provider
+    this.trackingProvider.init?.();
```

Comment:
do provide a fallback method which logs and handles error in case the trackingProvider is not initialized or injection issue happens